### PR TITLE
Support for connecting over TLS in Redis driver

### PR DESCRIPTION
**Pull request by ghik**

### CODE REVIEW

File: commons-redis/src/main/scala/com/avsystem/commons/redis/config/config.scala [hunk 1]
```diff
@@ -159,11 +163,21 @@ case class NodeConfig(
...
   maxWriteSizeHint: OptArg[Int] = 50000,
   reconnectionStrategy: RetryStrategy = immediately.andThen(exponentially(1.seconds)).maxDelay(8.seconds),
   debugListener: DebugListener = DevNullListener
 )
+
+/**
+  * Configuration options for TLS connection.
+  */
+case class TlsConfig(
+  sslContext: SSLContext = SSLContext.getDefault,
```
[the remark this answers is not in the document]
Comment (ghik) [reply commons-redis/src/main/scala/com/avsystem/commons/redis/config/config.scala:181]: Damnit, I was supposed to remove it.

[the remark this answers is not in the document]
Comment (kiemlicz) [reply commons-redis/src/main/scala/com/avsystem/commons/redis/config/config.scala]: `sslContext: SSLContext` - still, is this context supposed to be `init`ed or not?

[the remark this answers is not in the document]
Comment (ghik) [reply commons-redis/src/main/scala/com/avsystem/commons/redis/config/config.scala]: Yeah, I guess. But I simply forward it to Akka Streams which require it.

**Comment by ghik** [comment]
Benchmark results comparing Akka IO vs Akka Streams (TCP, not TLS):

```
branch 1.x (Akka IO)
[info] Benchmark                                                   Mode  Cnt        Score       Error  Units
[info] RedisClientBenchmark.connectionClientBatchBenchmark        thrpt   40  1201454.853 ± 12868.253  ops/s
[info] RedisClientBenchmark.connectionClientCommandBenchmark      thrpt   40   815007.259 ± 11000.403  ops/s
[info] RedisClientBenchmark.nodeClientBatchBenchmark              thrpt   40  1054810.149 ±  2146.605  ops/s
[info] RedisClientBenchmark.nodeClientCommandBenchmark            thrpt   40   763889.669 ±  9613.050  ops/s

branch redis-tls (Akka Streams)
[info] Benchmark                                                   Mode  Cnt        Score       Error  Units
[info] RedisClientBenchmark.connectionClientBatchBenchmark        thrpt   40  1190626.521 ± 10821.741  ops/s
[info] RedisClientBenchmark.connectionClientCommandBenchmark      thrpt   40   773835.843 ± 10810.744  ops/s
[info] RedisClientBenchmark.nodeClientBatchBenchmark              thrpt   40  1028292.512 ±  2016.765  ops/s
[info] RedisClientBenchmark.nodeClientCommandBenchmark            thrpt   40   737262.082 ±  9388.223  ops/s
```

**Comment by ghik** [comment]
I had to revert back to using Akka 2.5 because of dependency hell in downstream projects.

**Comment by ghik** [comment]
Finally, benchmark results for TLS:

```
branch redis-tls (TLS, Akka 2.5)
[info] Benchmark                                                   Mode  Cnt       Score       Error  Units
[info] RedisTlsClientBenchmark.connectionClientBatchBenchmark     thrpt   40  894993.534 ±  3405.239  ops/s
[info] RedisTlsClientBenchmark.connectionClientCommandBenchmark   thrpt   40  717074.312 ±  8626.959  ops/s
[info] RedisTlsClientBenchmark.nodeClientBatchBenchmark           thrpt   40  636770.374 ±  3013.325  ops/s
[info] RedisTlsClientBenchmark.nodeClientCommandBenchmark         thrpt   40  168636.986 ± 12579.763  ops/s
```

**Pull request merged**